dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/BackwardsReachingDefinitionsAnalysis.java

```diff
@@ -209,14 +213,17 @@ public boolean mayBeFinal() {
 
     private BackwardsReachingDefinitionsAnalysis() {}
 
-    private void initialize(Function function, Set<Register> finalRegisters) {
-        for (Event event : function.getEvents()) {
+    private void initializeWriterMap(Collection<Event> events) {
+        for (Event event : events) {
             if (event instanceof RegWriter writer) {
                 writerMap.put(writer, new Readers());
             }
         }
-        writerMap.put(null, new Readers());
-        for (Event event : function.getEvents()) {
+        writerMap.put(INITIAL_WRITER, new Readers());
+    }
```

Comment:
I don't like this change. I think the original request was not about saving a call to `function.getEvents`.
Just use `for (RegWriter writer : function.getEvents(RegWriter.class))`. You can even use `function.getEvents(RegWriter.class).foreach(w -> writerMap.put(w, new Readers())` if you prefer.